troubleshooting/ETW/Readme.md

```diff
@@ -127,43 +127,69 @@ This file will not exceed the configured max size and will be circularly overwri
 
 #### Configuration
 
-Configuration is controlled by a file named `ApplicationInsightsDiagnostics.json`.
+Configuration is controlled by a file named `ApplicationInsightsDiagnostics.json` or environment variables.
 The configuration file must be no more than 4 KiB, otherwise only the first 4 KiB of content will be read.
 
-**To enable self-diagnostics**, go to the [current working directory](https://en.wikipedia.org/wiki/Working_directory) of your process and create a configuration file.
+#### Two ways to enable self-diagnostics
+
+1. go to the [current working directory](https://en.wikipedia.org/wiki/Working_directory) of your process and create a configuration file.
 In most cases, you could just drop the file along your application.
-On Windows, you can use [Process Explorer](https://docs.microsoft.com/sysinternals/downloads/process-explorer), 
+On Windows, you can use [Process Explorer](https://docs.microsoft.com/sysinternals/downloads/process-explorer),
 double click on the process to pop up Properties dialog, and find "Current directory" in "Image" tab.
 Internally, the SDK looks for the configuration file located in [GetCurrentDirectory](https://docs.microsoft.com/dotnet/api/system.io.directory.getcurrentdirectory),
 and then [AppContext.BaseDirectory](https://docs.microsoft.com/dotnet/api/system.appcontext.basedirectory).
 You can also find the exact directory by calling these methods from your code.
 
-**To disable self-diagnostics**, delete the configuration file.
+    Example:
 
-Example: 
-```json
-{
-    "LogDirectory": ".",
-    "FileSize": 1024,
-    "LogLevel": "Error"
-}
-```
+    ```json
+    {
+        "LogDirectory": ".",
+        "FileSize": 1024,
+        "LogLevel": "Error"
+    }
+    ```
+
+2. For the case that users are not allowed to create the `ApplicationInsightsDiagnostics.json` config file
+at the above mentioning locations due to hosting enviornment limiations, configuration can be set in via enviornment variables.
+
+    Example:
+
+    ```csharp
+    {
+        // "LogDirectory" must be set if users want application insights to read configuration from envrionment variables. 
+        Environment.SetEnvironmentVariable("LogDirectory", ".");
+
+        // "FileSize" is an optional parameter, i.e. if "LogDirectory" was set, the default value for FileSize is 1 MiB.
+        Environment.SetEnvironmentVariable("FileSize", "1024");
+
+        // "LogLevel" is an optional parameter, i.e. if "LogDirectory" was set, the default value for LogLevel is LogAlways.
+        Environment.SetEnvironmentVariable("LogLevel", "LogAlways");
+    }
+    ```
+
+#### To disable self-diagnostics
+
+Depending on how self-diagnostcis was enabled, delete the configuration file
+or delete the enviornment variable(s).
```

Comment:
Might be always plural if you don't assign default value for missing parameters.

Reply:
I've updated to use a single environment variable to control the config. Thanks!